app/models/project.rb

```diff
@@ -415,6 +423,18 @@ def subscribed_repos(user)
     subscriptions.with_repository_subscription.where('repository_subscriptions.user_id = ?', user.id).map(&:repository).uniq
   end
 
+  def dependent_repos_view_query(limit, offset=0)
+    Repository.find_by_sql(["select * from repositories where id in (select repository_id from project_dependent_repositories where project_id = ? order by rank desc nulls last, stargazers_count desc limit ? offset ?);", id, limit, offset])
```

Comment:
I find it a lot easier to read SQL when the reserved words are in caps. Also can you split this into multiple lines?